### PR TITLE
[BugFix] schema id should default to index id

### PR DESCRIPTION
**Pull request by sduzh**
## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by sevev — APPROVED** [review]

**Review by sevev — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndexMeta.java [hunk 1]
```diff
@@ -89,18 +90,19 @@ public class MaterializedIndexMeta implements Writable, GsonPostProcessable {
     private boolean isColocateMVIndex = false;
 
     private Expr whereClause;
-    private Set<Long> updateSchemaBackendId = new HashSet<>();
+    private Set<Long> updateSchemaBackendId;
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndexMeta.java:93]: `private Set<Long> updateSchemaBackendId = new HashSet<>();`

will this not work for gson deserialize?

**Review by HangyuanLiu — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndexMeta.java (hunk 1 above)
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndexMeta.java:93]: Yes

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndexMeta.java:93]: how about create the Set in postGson? on-demand creating always introduces one level of indirection.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedIndexMeta.java:93]: This field has a quite low access frequency，so I also want to reduce the memory usage.

**Pull request merged**